astropy/table/typing.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
+
+from typing import (
+    TYPE_CHECKING,
+    Generic,
+    Protocol,
+    Self,
+    SupportsIndex,
+    TypeVar,
+    overload,
+)
+
+if TYPE_CHECKING:
+    import numpy as np
+    from numpy.typing import NDArray
+
+    from astropy.utils.data_info import MixinInfo
+
+__all__ = [
+    "MixinColumn",
+]
+
+
+T = TypeVar("T")
+
+
+class MixinColumn(Protocol, Generic[T]):
```

Comment:
Protocols serve as Generic

```suggestion
class MixinColumn(Protocol[T]):
```

Reply:
how wow, I didn't know, thanks !